common/src/main/java/com/wynntils/mc/utils/ComponentUtils.java

```diff
@@ -6,81 +6,117 @@
+                handleStyleDifference(oldStyle, style, result);
+                result.append(string);
+
+                oldStyle = style;
+
+                return Optional.empty();
+            }
+        }, Style.EMPTY);
+
+        return result.toString();
+    }
+
     public static String getUnformatted(Component msg) {
         return msg.getString();
     }
 
-    public static String fromComponent(Component component) {
-        StringBuilder result = new StringBuilder();
-        Style oldStyle = Style.EMPTY;
-
-        component.visit(
-                (style, string) -> {
-                    if (!oldStyle.isEmpty()) {
-                        StringBuilder different =
-                                getToAdd(oldStyle, style); // Try to get difference to add
-                        if (different != null) {
-                            result.append(different);
-                            return Optional.empty();
-                        }
-
-                        result.append(ChatFormatting.RESET);
-                    }
-
-                    if (style.getColor() != null) {
-                        Optional<ChatFormatting> color = getChatFormatting(style.getColor());
-                        color.ifPresent(result::append);
-                    }
-
-                    if (style.isBold()) result.append(ChatFormatting.BOLD);
-                    if (style.isItalic()) result.append(ChatFormatting.ITALIC);
-                    if (style.isUnderlined()) result.append(ChatFormatting.UNDERLINE);
-                    if (style.isStrikethrough()) result.append(ChatFormatting.STRIKETHROUGH);
-                    if (style.isObfuscated()) result.append(ChatFormatting.OBFUSCATED);
-
-                    return Optional.empty(); // dont break
-                },
-                Style.EMPTY);
+    @Nullable
+    public static String getFormatted(String loreString) {
+        MutableComponent component = Component.Serializer.fromJson(loreString);
+        if (component == null) return null;
 
-        return result.toString();
+        return ComponentUtils.getFormatted(component);
     }
 
-    private static StringBuilder getToAdd(Style oldStyle, Style newStyle) {
-        int oldColorInt =
-                Optional.ofNullable(oldStyle.getColor()).map(TextColor::getValue).orElse(-1);
-        int newColorInt =
-                Optional.ofNullable(newStyle.getColor()).map(TextColor::getValue).orElse(-1);
+    @Nullable
+    public static String getUnformatted(String loreString) {
+        MutableComponent component = Component.Serializer.fromJson(loreString);
+        if (component == null) return null;
 
-        StringBuilder add = new StringBuilder();
+        return component.getString();
+    }
 
-        if (oldColorInt != newColorInt) return null;
+        private static void handleStyleDifference(Style oldStyle, Style newStyle, StringBuilder result) { //style could've changed
```

Comment:
A bit odd placement of the comment. I'm not sure what it's even supposed to mean.

Reply:
Will expand on the comment later